src/Base/UnitsSchemaImperial1.cpp

```diff
@@ -205,70 +205,107 @@ QString UnitsSchemaImperialDecimal::schemaTranslate(const Base::Quantity& quant,
 QString UnitsSchemaImperialBuilding::schemaTranslate(const Quantity &quant, double &factor, QString &unitString)
 {
     // this schema expresses distances in feet + inches + fractions
-    // ex: 3'- 4 1/4"
+    // ex: 3'- 4 1/4" with proper rounding
     Unit unit = quant.getUnit();
     if (unit == Unit::Length) {
         unitString = QString::fromLatin1("in");
         factor = 25.4;
-        double inchValue = std::abs(quant.getValue())/25.4;
-        int feet = inchValue/12;
-        double inchPart = inchValue - (double)feet*12;
-        int inches = (int)inchPart;
-        double fraction = inchPart - (int)inchPart;
-        if (fraction > 0.9375) {
-            inches++;
-            fraction = 0.0;
-        }
 
-        // if the quantity is too small it is rounded to zero
-        if (std::abs(quant.getValue()) <= 1.5875)
+        // Total number of inches to format
+        double totalInches = std::abs(quant.getValue())/factor;
+        
+        // minimum denominator (8 for 1/8, 16 for 1/16, etc)
+        int       minden; 
+
+        // Outputs
+        int              feet;    // whole feet
+        int              inches;  // whole inches
+        int              num,den; // numerator and denominator of fractional val

```

Comment:
formatting: looks like there is an extra tab for the above lines.

Reply:
Not tabs, just spaces. Cleaned it up though.